fdbserver/DataDistribution.actor.cpp

```diff
@@ -3106,7 +3110,20 @@ ACTOR Future<Void> teamTracker(DDTeamCollection* self, Reference<TCTeamInfo> tea
 			ASSERT( self->optimalTeamCount >= 0 );
 			self->zeroOptimalTeams.set(self->optimalTeamCount == 0);
 		}
-		throw;
+		try {
+			bool containsFailed = teamContainsFailedServer(self, team);
+			// If server is marked as failed, a lost race condition could result in RelocateShards not being sent
+			// This is implemented as a safety net so we do not have orphaned shards sitting around
+			if (containsFailed && e.code() == error_code_actor_cancelled) {
```

Comment:
It is a bit faster if the order of `e.code() == error_code_actor_cancelled` and `containsFailed ` is switched.